### PR TITLE
SPU2-X: Fix Project Zero 2 regression

### PR DESCRIPTION
**Pull request by FlatOutPS2**
Fixes regression that caused audio stuttering when playing voices at some locations in Project Zero 2.

Chaos Legion seems unaffected.

Closes: https://github.com/PCSX2/pcsx2/issues/2314

@PCSX2/bug-squad Please test this on other games. Check for issues with voices/music stuttering or repeating.

### CODE REVIEW

**Comment by refractionpcsx2** [comment]
Just out of curiosity, if you put back in the >= TSA, does that break Chaos Legion? it would make more sense if it triggers when it was equal too, unless it only triggers once it passes it.

**Comment by FlatOutPS2** [comment]
> Just out of curiosity, if you put back in the >= TSA, does that break Chaos Legion? it would make more sense if it triggers when it was equal too, unless it only triggers once it passes it.

Yes, TSA breaks Chaos Legion. TDA breaks PZ2.

**Comment by refractionpcsx2** [comment]
interesting, I wonder why that is.  I guess it has to be inside it then :P

**Comment by lightningterror** [comment]
> interesting, I wonder why that is. I guess it has to be inside it then :P

Voodoo ?

**Comment by refractionpcsx2** [comment]
> Voodoo ?

Could be  :P it's more likely something like the memory read has to have passed the IRQ Address for it to trigger, rather than it firing when it's directly on it.

File: plugins/spu2-x/src/Dma.cpp [hunk 1]
```diff
@@ -291,14 +291,14 @@ void V_Core::PlainDMAWrite(u16 *pMem, u32 size)
         // Buffer doesn't wrap/overflow!
         // Just set the TDA and check for an IRQ...
 
-        TDA = (buff1end + 1) & 0xfffff;
+        TDA = buff1end;
```
Comment (turtleli) [inline plugins/spu2-x/src/Dma.cpp]: Is this change necessary too? It makes the TDA handling inconsistent (see code above).

**Comment by refractionpcsx2** [comment]
Why did you close this?

**Comment by FlatOutPS2** [comment]
The code was meant for testing not reviewing. No one seemed interested in testing this, and there is something else wrong in the code. So I'll close this while I investigate if that is at the root of this issue as well.

I actually didn't manually close it myself. I send an empty commit, to see if it would automatically close, and it did. :smile:

**Comment by refractionpcsx2** [comment]
Fair enough :) Yeah im sure we would have tested it but of course we wanna make sure there's no problems with the code too so we're not just bug finding and actually testing the changes intended :)

Looking forward to PR Mk 2 :P
